tests/browser/elements.js

```diff
@@ -180,3 +180,26 @@ test('allow objects to be passed', function (t) {
   t.ok(result.outerHTML.indexOf('<div foo="bar">hey</div>') !== -1, 'contains foo="bar"')
   t.end()
 })
+
+test('supports extended build-in elements', function (t) {
+  t.plan(1)
+
+  var originalCreateElement = document.createElement
+  var optionsArg
+
+  // this iife is a must to avoid illegal invocation type errors, caused by transformed nanohtml tests
+  (function () {
+    document.createElement = function () {
+      optionsArg = arguments[1]
+      return originalCreateElement.apply(this, arguments)
+    }
+  })()
+
+  ;html`<div is="my-div"></div>`
+
+  t.ok(typeof optionsArg === 'object' && optionsArg.is === 'my-div', 'properly passes optional extends object')
+
+  delete document.createElement
```

Comment:
`document.createElement` should be reset to `originalCreateElement` here i think? else future tests below this one won't be able to use it.

Reply:
@goto-bus-stop 
You absolutely right.
Actaully this is what I want to archieve by `delete document.createElement`. It's a bit opaque though.
But let me explain.
1) the original `createElement` method sits in the [`HTMLDocument.prototype` object](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDocument) of the [Document interface](https://developer.mozilla.org/en-US/docs/Web/API/Document)
2) I cache [`document.createElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement) from there
3) Now here is the important bit, I store a new `createElement` method at the `document` instance directly, shadowing the original at it's prototype below
4) that's why deleting it should be sufficient as I understand it

Are you fine with that?
Else I'm also okay to explicitly reassign the original.

Reply:
oh i see, that makes sense. i'd like a comment to clarify that though. something short like `// revert to prototype method` might be enough

Reply:
awesome idea, just did it 💪